pkg/proxy/endpointslicecache.go

```diff
@@ -124,7 +124,12 @@ func (cache *EndpointSliceCache) EndpointsMap(serviceNN types.NamespacedName) En
 func (cache *EndpointSliceCache) endpointInfoByServicePort(serviceNN types.NamespacedName) map[ServicePortName]map[string]Endpoint {
 	endpointInfoBySP := map[ServicePortName]map[string]Endpoint{}
 
-	for _, sliceInfo := range cache.sliceByServiceMap[serviceNN] {
+	sliceInfoByName, ok := cache.sliceByServiceMap[serviceNN]
```

Comment:
add a type decl for the map map
`type ServicePortToEndpointMap map[ServicePortName]map[string]Endpoint`
prefer to move var decl closer to where used:

```
sliceInfoByName, ok := cache.sliceByServiceMap[serviceNN]
 

 
add a type decl for the map map
`type spToEndpointMap map[ServicePortName]map[string]Endpoint`
prefer to move var decl closer to where used:

```

```
if !ok {
  return spToEndpointMap{}
}
endpointInfoBySP := map[ServicePortName]map[string]Endpoint{}
```

Reply:
Good call, thanks, should be fixed now.